jetty-core/jetty-siwe/src/main/java/org/eclipse/jetty/security/siwe/EthereumAuthenticator.java

```diff
@@ -0,0 +1,771 @@
+import org.eclipse.jetty.security.siwe.internal.SignInWithEthereumParser;
+import org.eclipse.jetty.security.siwe.internal.SignInWithEthereumToken;
+import org.eclipse.jetty.server.FormFields;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.Response;
+import org.eclipse.jetty.server.Session;
+import org.eclipse.jetty.util.Blocker;
+import org.eclipse.jetty.util.BufferUtil;
+import org.eclipse.jetty.util.Callback;
+import org.eclipse.jetty.util.CharsetStringBuilder.Iso88591StringBuilder;
+import org.eclipse.jetty.util.Fields;
+import org.eclipse.jetty.util.IncludeExcludeSet;
+import org.eclipse.jetty.util.URIUtil;
+import org.eclipse.jetty.util.UrlEncoded;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.eclipse.jetty.server.FormFields.getFormEncodedCharset;
+
+public class EthereumAuthenticator extends LoginAuthenticator
+{
+    private static final Logger LOG = LoggerFactory.getLogger(EthereumAuthenticator.class);
+
+    public static final String LOGIN_PATH_PARAM = "org.eclipse.jetty.security.siwe.login_path";
+    public static final String AUTHENTICATION_PATH_PARAM = "org.eclipse.jetty.security.siwe.authentication_path";
+    public static final String NONCE_PATH_PARAM = "org.eclipse.jetty.security.siwe.nonce_path";
+    public static final String MAX_MESSAGE_SIZE_PARAM = "org.eclipse.jetty.security.siwe.max_message_size";
+    public static final String LOGOUT_REDIRECT_PARAM = "org.eclipse.jetty.security.siwe.logout_redirect_path";
+    public static final String DISPATCH_PARAM = "org.eclipse.jetty.security.siwe.dispatch";
+    public static final String ERROR_PAGE = "org.eclipse.jetty.security.siwe.error_page";
+    public static final String J_URI = "org.eclipse.jetty.security.siwe.URI";
+    public static final String J_POST = "org.eclipse.jetty.security.siwe.POST";
+    public static final String J_METHOD = "org.eclipse.jetty.security.siwe.METHOD";
+    public static final String ERROR_PARAMETER = "error_description_jetty";
+    private static final String DEFAULT_AUTHENTICATION_PATH = "/auth/login";
+    private static final String DEFAULT_NONCE_PATH = "/auth/nonce";
+    private static final String NONCE_SET_ATTR = "org.eclipse.jetty.security.siwe.nonce";
+
+    private final IncludeExcludeSet<String, String> _chainIds = new IncludeExcludeSet<>();
+    private final IncludeExcludeSet<String, String> _schemes = new IncludeExcludeSet<>();
+    private final IncludeExcludeSet<String, String> _domains = new IncludeExcludeSet<>();
+
+    private String _loginPath;
+    private String _authenticationPath = DEFAULT_AUTHENTICATION_PATH;
+    private String _noncePath = DEFAULT_NONCE_PATH;
+    private long _maxMessageSize = 4 * 1024;
+    private String _logoutRedirectPath;
+    private String _errorPage;
+    private String _errorPath;
+    private String _errorQuery;
+    private boolean _dispatch;
+
+    public EthereumAuthenticator()
+    {
+    }
```

Comment:
Add a LOG.warning here that this is experimental code.

Reply:
Done I put the warning `Sign-In With Ethereum support is experimental and not suited for production use`.

But I don't see the point of adding this module with a warning not to use it. It doesn't really need to be experimental.

Reply:
OK, if you are happy to support in production, then remove!